File: Python/Colleen.py
# Self Printing Python

def printsource():
	source = [
		'# Self Printing Python',
		'',
		'def printsource():',
		'	source = [',
		'	]',
		'	for x in range(0, 4):',
		'		print(source[x])',
		'	for x in range(0, 18):',
		'		print("		{0}{1}{2},".format(chr(39), source[x], chr(39)))',
		'	for x in range(4, 18):',
		'		print(source[x])',
		'',
		'def main():',
		'	# This is a comment in python',
		'	printsource()',
		'',
		'if __name__ == "__main__":',
		'	main()',
	]
	for x in range(0, 4):
		print(source[x])
	for x in range(0, 18):
		print("		{0}{1}{2},".format(chr(39), source[x], chr(39)))
	for x in range(4, 18):
		print(source[x])

def main():
	# This is a comment in python
	printsource()

if __name__ == "__main__":
	main()

File: Ruby/Colleen.rb
# Self Printing Ruby

def print
	source = [
		'# Self Printing Ruby',
		'',
		'def print',
		'	source = [',
		'	]',
		'	for x in 0..3',
		'		puts source[x]',
		'	end',
		'	q = 39',
		'	for x in 0..22',
		'		puts "		#{q.chr}#{source[x]}#{q.chr},"',
		'	end',
		'	for x in 4..22',
		'		puts source[x]',
		'	end',
		'end',
		'',
		'def main',
		'	# This is a comment in ruby',
		'	print',
		'end',
		'',
		'main()',
	]
	for x in 0..3
		puts source[x]
	end
	q = 39
	for x in 0..22
		puts "		#{q.chr}#{source[x]}#{q.chr},"
	end
	for x in 4..22
		puts source[x]
	end
end

def main
	# This is a comment in ruby
	print
end

main()
